darknet_ros/CMakeLists.txt

```diff
@@ -46,99 +96,99 @@ if ( X11_FOUND )
 )
 
 set(PROJECT_LIB_FILES
     src/YoloObjectDetector.cpp                    src/image_interface.cpp
+    src/yolo_object_detector_node.cpp
 )
 
 set(DARKNET_CORE_FILES
-    ${DARKNET_PATH}/src/activation_layer.c        ${DARKNET_PATH}/src/im2col.c
-    ${DARKNET_PATH}/src/activations.c             ${DARKNET_PATH}/src/image.c
-    ${DARKNET_PATH}/src/avgpool_layer.c           ${DARKNET_PATH}/src/layer.c
-    ${DARKNET_PATH}/src/batchnorm_layer.c         ${DARKNET_PATH}/src/list.c
-    ${DARKNET_PATH}/src/blas.c                    ${DARKNET_PATH}/src/local_layer.c
-    ${DARKNET_PATH}/src/box.c                     ${DARKNET_PATH}/src/lstm_layer.c
-    ${DARKNET_PATH}/src/col2im.c                  ${DARKNET_PATH}/src/matrix.c
-    ${DARKNET_PATH}/src/connected_layer.c         ${DARKNET_PATH}/src/maxpool_layer.c
-    ${DARKNET_PATH}/src/convolutional_layer.c     ${DARKNET_PATH}/src/network.c
-    ${DARKNET_PATH}/src/cost_layer.c              ${DARKNET_PATH}/src/normalization_layer.c
-    ${DARKNET_PATH}/src/crnn_layer.c              ${DARKNET_PATH}/src/option_list.c
-    ${DARKNET_PATH}/src/crop_layer.c              ${DARKNET_PATH}/src/parser.c
-    ${DARKNET_PATH}/src/cuda.c                    ${DARKNET_PATH}/src/region_layer.c
-    ${DARKNET_PATH}/src/data.c                    ${DARKNET_PATH}/src/reorg_layer.c
-    ${DARKNET_PATH}/src/deconvolutional_layer.c   ${DARKNET_PATH}/src/rnn_layer.c
-    ${DARKNET_PATH}/src/demo.c                    ${DARKNET_PATH}/src/route_layer.c
-    ${DARKNET_PATH}/src/detection_layer.c         ${DARKNET_PATH}/src/shortcut_layer.c
-    ${DARKNET_PATH}/src/dropout_layer.c           ${DARKNET_PATH}/src/softmax_layer.c
-    ${DARKNET_PATH}/src/gemm.c                    ${DARKNET_PATH}/src/tree.c
-    ${DARKNET_PATH}/src/gru_layer.c               ${DARKNET_PATH}/src/utils.c
-    ${DARKNET_PATH}/src/upsample_layer.c          ${DARKNET_PATH}/src/logistic_layer.c
-    ${DARKNET_PATH}/src/l2norm_layer.c            ${DARKNET_PATH}/src/yolo_layer.c
-    ${DARKNET_PATH}/src/iseg_layer.c              ${DARKNET_PATH}/src/image_opencv.cpp
-
-    ${DARKNET_PATH}/examples/art.c                ${DARKNET_PATH}/examples/lsd.c
-    ${DARKNET_PATH}/examples/nightmare.c          ${DARKNET_PATH}/examples/instance-segmenter.c
-    ${DARKNET_PATH}/examples/captcha.c            ${DARKNET_PATH}/examples/regressor.c
-    ${DARKNET_PATH}/examples/cifar.c              ${DARKNET_PATH}/examples/rnn.c
-    ${DARKNET_PATH}/examples/classifier.c         ${DARKNET_PATH}/examples/segmenter.c
-    ${DARKNET_PATH}/examples/coco.c               ${DARKNET_PATH}/examples/super.c
-    ${DARKNET_PATH}/examples/darknet.c            ${DARKNET_PATH}/examples/tag.c
-    ${DARKNET_PATH}/examples/detector.c           ${DARKNET_PATH}/examples/yolo.c
-    ${DARKNET_PATH}/examples/go.c
+    ${DARKNET_PATH}/src/activation_layer.c        ${DARKNET_PATH}/src/activations.c
+    ${DARKNET_PATH}/src/art.c                     ${DARKNET_PATH}/src/avgpool_layer.c
```

Comment:
art.c is not in the darknet submodule referred to in this repo, do you know what causes this @Ar-Ray-code ?

Reply:
Because the current submodule of darknet_ros is [pjreddie/darknet](https://github.com/pjreddie/darknet), but in my branch it refers to [AlexeyAB/darknet](https://github.com/AlexeyAB/darknet) because my branch refers to it.

Reply:
> Did you devel on top of the master or the foxy branch, @Ar-Ray-code? Could you rebase such that only your changes are included?

Maybe I did not understand the meaning of this question.
This pull request is built on ROS2-Foxy, but many robotics engineers use ROS1-Noetic...

Reply:
Thanks for your explanation. So your PR should be merged into leggedrobotics/darknet_ros/foxy

Reply:
I compiled your code, seems to work fine on ROS2 Foxy, but I didn't get it to run on the GPU yet, even though CUDA was found during building. Do you set something specific to have darknet_ros run on your GPU instead of CPU?

Reply:
Sorry, the CUDA setting was turned off.

CMakeLists.txt describes the settings to turn on/off CUDA and cuDNN for speed comparison.
Please change `set(CUDA_ENABLE OFF)` to `set(CUDA_ENABLE ON)`.

https://github.com/Ar-Ray-code/darknet_ros/blob/master/darknet_ros/CMakeLists.txt#L10